From: George McCollister <george.mccollister@gmail.com>
Date: Mon, 26 Mar 2012 10:06:49 -0500
Subject: [PATCH] don't fillout installed-size for postgresql armeb

Since postgresql goes on a different partition on the OrionLX the free
disk space checks are irrelevant since they are made against the root
partition.

Signed-off-by: George McCollister <george.mccollister@gmail.com>
---
 opkg.py | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/opkg.py b/opkg.py
index 2531871..f8c7261 100644
--- a/opkg.py
+++ b/opkg.py
@@ -162,9 +162,10 @@ class Package:
             ## sys.stderr.write("  extracting control.tar.gz from %s\n"% (fn,)) 
 
             ar = arfile.ArFile(f)
-            tarStream = ar.open("data.tar.gz")
-            tarf = tarfile.open("data.tar.gz", "r", tarStream)
-            self.installed_size = sum([f.size for f in tarf.getmembers()])
+	    if not (self.filename.startswith("postgresql") and self.filename.endswith("armeb.ipk")):
+                tarStream = ar.open("data.tar.gz")
+                tarf = tarfile.open("data.tar.gz", "r", tarStream)
+                self.installed_size = sum([f.size for f in tarf.getmembers()])
 
             tarStream = ar.open("control.tar.gz")
             tarf = tarfile.open("control.tar.gz", "r", tarStream)
